src/state/session/agent.ts

```diff
@@ -174,31 +188,20 @@ export async function createAgentAndCreateAccount(
     logger.error(e, {context: `session: failed snoozeEmailConfirmationPrompt`})
   }
 
-  return prepareAgent(agent, gates, moderation, onSessionChange)
+  return prepareAgent(agent, gates, moderation)
 }
 
 async function prepareAgent(
   agent: BskyAgent,
   // Not awaited in the calling code so we can delay blocking on them.
   gates: Promise<void>,
   moderation: Promise<void>,
-  onSessionChange: (
-    agent: BskyAgent,
-    did: string,
-    event: AtpSessionEvent,
-  ) => void,
 ) {
   // There's nothing else left to do, so block on them here.
   await Promise.all([gates, moderation])
 
   // Now the agent is ready.
   const account = agentToSessionAccountOrThrow(agent)
-  agent.setPersistSessionHandler(event => {
-    onSessionChange(agent, account.did, event)
```

Comment:
The intention in the original code is to "decide" on a `did` by this point (note we're reading it from the account) and then to consider the account fully tied to that `did`. This is why we're using `account.did` (which is immutable) and not reading from the `session` object. I think we'd want to keep a similar guarantee after restructuring.

Reply:
Going to tweak with this in a separate PR per discussion at start of week. Left some context in slack, but here I've just duplicated the previous behavior (as far as I can tell) by using a weird hack.